repair/repair.cc

```diff
@@ -2086,7 +2083,7 @@ future<> repair_service::repair_tablets(repair_uniq_id rid, sstring keyspace_nam
                     found = true;
                     break;
                 }
-                if (primary_replica_only) {
+                if (primary_replica_only && ranges_specified.empty()) {
```

Comment:
Better to have the caller send a full range. Instead of special casing empty as everything.

Reply:
I added some comments to clarify it.

Reply:
By "full range" I meant (-inf, +inf).

Reply:
Yes, but full range is even worse because the user is not really asking to repair full ranges. The full range thing is only useful for the purpose of intersecting calculation. That's it. It is not making the real intention more clear.